stdlib/Test/src/Test.jl

```diff
@@ -601,16 +601,33 @@ macro test_throws(extype, ex)
     :(do_test_throws($result, $orig_ex, $(esc(extype))))
 end
 
+const MACROEXPAND_LIKE = Symbol.(("@macroexpand", "@macroexpand1", "macroexpand"))
+
 # An internal function, called by the code generated by @test_throws
 # to evaluate and catch the thrown exception - if it exists
 function do_test_throws(result::ExecutionResult, orig_expr, extype)
     if isa(result, Threw)
         # Check that the right type of exception was thrown
         success = false
         exc = result.exception
+        # NB: Throwing LoadError from macroexpands is deprecated, but in order to limit
+        # the breakage in package tests we add extra logic here.
+        from_macroexpand =
+            orig_expr isa Expr &&
+            orig_expr.head in (:call, :macrocall) &&
+            orig_expr.args[1] in MACROEXPAND_LIKE
         if isa(extype, Type)
-            success = isa(exc, extype)
+            success =
+                if from_macroexpand && extype == LoadError && exc isa Exception
+                    Base.depwarn("macroexpands no longer throw a LoadError. @test_throws passed without checking the error type!", Symbol("@test_throws LoadError [@]macroexpand[1]"))
```

Comment:
`@test_deprecated` expects the message passed to `depwarn()` to contain the string "deprecated". This is a bit unfortunate in this case, but we can reword a little without it becoming too awkward.

Also, I think need to use :do_test_throws as the second argument here so that the caller of the deprecation will be found correctly.

```suggestion
                    Base.depwarn("macroexpand no longer throw a LoadError so `@test_throws LoadError ...` is deprecated and passed without checking the error type!", :do_test_throws)
```